eclipse/src/saros/ui/actions/ChangeXMPPAccountAction.java

```diff
@@ -70,15 +82,29 @@ public ChangeXMPPAccountAction() {
     connectionHandler.addConnectionStateListener(connectionStateListener);
     setMenuCreator(this);
     updateStatus(connectionHandler.getConnectionState());
+
+    accountService.addListener(accountStoreListener);
   }
 
   @Override
   public void run() {
 
+    if (connectionHandler.isConnected()) {
+      XMPPConnectionSupport.getInstance().disconnect();
+      return;
+    }
+
+    final XMPPAccount lastUsedAccount = XMPPConnectionSupport.getInstance().getCurrentXMPPAccount();
+
+    final List<XMPPAccount> accounts = accountService.getAllAccounts();
+
+    final boolean exists = accounts.indexOf(lastUsedAccount) != -1;
```

Comment:
No it does not work like that because you can query accounts from the store. You can store them, however thry may become invalid  and this is what the logic checks.